luxonis_train/utils/config.py

```diff
@@ -77,6 +77,23 @@ class ModelConfig(BaseModelExtraForbid):
     visualizers: list[AttachedModuleConfig] = []
     outputs: list[str] = []
 
+    @model_validator(mode="after")
+    def check_main_metric(self) -> Self:
+        for metric in self.metrics:
+            if metric.is_main_metric:
+                logger.info(f"Main metric: `{metric.name}`")
+                return self
+
+        logger.warning("No main metric specified.")
+        if self.metrics:
+            metric = self.metrics[0]
+            metric.is_main_metric = True
+            name = metric.alias or metric.name
+            logger.info(f"Setting '{name}' as main metric.")
+        else:
+            logger.error("No metrics specified. This is likely unintended.")
```

Comment:
Do we expect any crashes if no metrics is specified or is this handled correctly?

Reply:
No, it would just mean we can't use checkpoints based on metrics. And it can also be a valid when the config is only for export/archiving etc

Reply:
But I changed the log message to reflect this valid use case.